From 59254fef430dc31a8ea92c2815be75ec0bbfd15d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dawid=20Wr=C3=B3bel?= <me@dawidwrobel.com>
Date: Thu, 22 Oct 2026 13:39:01 +0200
Subject: [PATCH] Update 12+13-feature-linux-systray-incl-minimise.patch to
 register the Linux systray integration on all architectures (#623)

Thunderbird registers its Rust system tray component only on x86_64
Linux, because the Rust tray depends on the ksni crate only for that
target. Our systray patch replaces that component with
nsMessengerUnixIntegration which builds fine on all architectures.
---
 .../12+13-feature-linux-systray-incl-minimise.patch | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/153/features/12+13-feature-linux-systray-incl-minimise.patch b/153/features/12+13-feature-linux-systray-incl-minimise.patch
index f8134233..500c1c88 100644
--- a/153/features/12+13-feature-linux-systray-incl-minimise.patch
+++ b/153/features/12+13-feature-linux-systray-incl-minimise.patch
@@ -419,9 +419,16 @@ diff --git a/mailnews/base/src/MailNotificationManager.sys.mjs b/mailnews/base/s
 diff --git a/mailnews/base/src/components.conf b/mailnews/base/src/components.conf
 --- a/mailnews/base/src/components.conf
 +++ b/mailnews/base/src/components.conf
-@@ -325,10 +325,11 @@ elif (
-     and buildconfig.substs["TARGET_CPU"] == "x86_64"
- ):
+@@ -319,16 +319,14 @@ elif buildconfig.substs["OS_ARCH"] == "WINNT":
+             "type": "nsMessengerWinIntegration",
+             "headers": ["/comm/mailnews/base/src/nsMessengerWinIntegration.h"],
+         },
+     ]
+-elif (
+-    buildconfig.substs["OS_ARCH"] == "Linux"
+-    and buildconfig.substs["TARGET_CPU"] == "x86_64"
+-):
++elif buildconfig.substs["OS_ARCH"] == "Linux":
      Classes += [
          {
 -            "cid": "{d9e45fee-1dd1-11b2-938c-9147855ed837}",
-- 
2.56.0

